src/test/java/com/elytradev/concrete/recipe/ShapedRecipeTest.java

```diff
@@ -0,0 +1,126 @@
+/*
```

Comment:
Should this be moved to a test package? I'm not exactly sure how JUnit tests are meant to be organized, but it seems like it'd be better to (at some point) relocate this.

Reply:
It's in a test source set, which is enough. (Note the path: src/**test**/java)